libcxx/test/libcxx/diagnostics/cmath.nodiscard.verify.cpp

```diff
@@ -6,162 +6,477 @@
+  std::log1p(0.L); // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+  std::log1p(0U);  // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+
+  std::log2(0.F); // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+  std::log2(0.);  // expected-warning-re 0-1 {{ignoring return value of function declared with {{.*}} attribute}}
+  std::log2(0.L); // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+  std::log2(0U);  // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+
+  std::logb(0.F); // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+  std::logb(0.);  // expected-warning-re 0-1 {{ignoring return value of function declared with {{.*}} attribute}}
+  std::logb(0.L); // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+  std::logb(0U);  // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+
+  // Basic operations
+
+  std::fmax(0.F, 0.F); // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+  std::fmax(0., 0.);   // expected-warning-re 0-1 {{ignoring return value of function declared with {{.*}} attribute}}
+  std::fmax(0.L, 0.L); // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+  std::fmax(0U, 0U);   // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+
+  std::fmin(0.F, 0.F); // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+  std::fmin(0., 0.);   // expected-warning-re 0-1 {{ignoring return value of function declared with {{.*}} attribute}}
+  std::fmin(0.L, 0.L); // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+  std::fmin(0U, 0U);   // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+
+  std::fmod(0.F, 0.F); // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+  std::fmod(0., 0.);   // expected-warning-re 0-1 {{ignoring return value of function declared with {{.*}} attribute}}
+  std::fmod(0.L, 0.L); // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+  std::fmod(0U, 0U);   // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+
+  std::modf(0.F, 0); // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+  std::modf(0., 0);  // expected-warning-re 0-1 {{ignoring return value of function declared with {{.*}} attribute}}
+  std::modf(0.L, 0); // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
```

Comment:
Perhaps it's also reasonable to introduce `float`, `double`, and `long double` variables for these. Although this looks a bit messy to me.